src/Solve.jl

```diff
@@ -81,65 +125,67 @@ function _can_solve_internal(A::MatElem{T}, b::MatElem{T}, task::Symbol; side::S
     throw(ArgumentError("Unsupported argument :$side for side: Must be :left or :right."))
   end
 
-  isright = side === :right
+  isleft = side === :left
 
   R = base_ring(A)
 
-  # Build [ A | b ] and compute rref
-  if isright
-    nrows(A) != nrows(b) && error("Incompatible matrices")
-    mu = hcat(A, b)
-    ncolsA = ncols(A)
-    ncolsb = ncols(b)
-  else
-    ncols(A) != ncols(b) && error("Incompatible matrices")
-    mu = _hcat_transposed(A, b)
-    # For side == :left we pretend that A and b are transposed!
-    ncolsA = nrows(A)
-    ncolsb = nrows(b)
+  if isleft
+    # For side == :left, we pretend that A and b are transposed
+    A = lazy_transpose(A)
+    b = lazy_transpose(b)
```

Comment:
overwriting these already existing variables with something of a different type is unfortunately type unstable